docs/spec/spec.md

```diff
@@ -3184,5 +3195,132 @@ service on new graphql:Listener(9090) {
+4. Annotate the `loadXxx` method written in step two with the `@dataloader:Loader` annotation and pass the required configuration.
+
+The following sections provide further explanation for each of the above steps.
+
+#### 15.1.2.1. `loadXxx` Method Mapping
+
+To engage the DataLoader with a GraphQL field (let's assume the field name is `xxx`), the user must define a corresponding `loadXxx` method, where `Xxx` represents the Pascal-cased name of the GraphQL field. The `loadXxx` method and the `xxx` method should have the same resource accessor or should be remote methods. The `loadXxx` method can include some or all of the parameters from the GraphQL field, as well as the `map<dataloader:DataLoader>` parameter. Adding the parameters of the GraphQL field to the `loadXxx` method is optional. However, if these
+
+ parameters are added, the GraphQL Engine will make the same parameter values of the GraphQL field available to the `loadXxx` method.
+
+The user is responsible for implementing the logic to collect the keys of the data to be loaded into the `DataLoader` in the `loadXxx` method. The GraphQL Engine guarantees the execution of the `loadXxx` method prior to the `xxx` method. Subsequently, the user can implement the logic to retrieve the data from the `DataLoader` within the `xxx` method.
+
+#### 15.1.2.2. The `@dataloader:Loader` Annotation
+
+The `@dataloader:Loader` annotation is used to configure the `loadXxx` method. It serves the purpose of avoiding the addition of `loadXxx` as a field in the GraphQL schema. This annotation accepts a map of batch load functions to be used by the `DataLoader`.
+
+The map consists of unique identifiers as keys and user-written batch load functions as values. Each unique key represents a separate `DataLoader` object. When the GraphQL Engine encounters the `@dataloader:Loader` annotation, it uses the provided configuration to create `DataLoader` objects for each unique key.
+
+Using the same unique key across multiple `@dataloader:Loader` annotations will result in the same `DataLoader` object being shared by multiple GraphQL fields. This enables efficient batching and caching of data loading operations. The following is the type definition of this annotation.
+
+```ballerina
+# Provides configurations for the DataLoaders.
+# + batchFunctions - The map of batch function to be used in the DataLoader
+public type LoaderConfig record {|
+    map<(isolated function (readonly & anydata[] keys) returns anydata[]|error)> batchFunctions;
+|};
+
+# The annotation to configure the load method with batch functions.
+public annotation LoaderConfig Loader on object function;
+```
```

Comment:
Do we need to add these to the spec? May be we can add an example instead of this.

Reply:
Added an example at the end. Added definition here to show the function type of 
a batch load function.

 User can choose to override the return type of the load method with a narrowed type. I think adding just an example may not represent the correct type of batch load method.

Reply:
Will move this to the annotations section